fem/prestriction.hpp

```diff
@@ -68,6 +68,20 @@ class ParNCH1FaceRestriction : public H1FaceRestriction
        @param[in,out] y The L-vector degrees of freedom. */
    void AddMultTranspose(const Vector &x, Vector &y) const override;
 
+   /** @brief Gather the degrees of freedom, i.e. goes from face E-Vector to
+       L-Vector.
+
+       @param[in,out]  x The face E-Vector degrees of freedom with the given format:
+                         face_dofs x vdim x nf
+                         where nf is the number of interior or boundary faces
+                         requested by @a type in the constructor.
+                         The face_dofs should be ordered according to the given
+                         ElementDofOrdering.
+       @param[in,out] y The L-vector degrees of freedom.
+
+       @note @a x is used for computation. */
+   void AddMultTranspose(Vector &x, Vector &y) const override;
```

Comment:
If I understand correctly, this is a version of `AddMultTranspose` that does something in-place or otherwise destructive that overwrites the `x` vector. That should probably be documented explicitly, and I think it should probably also be given a different name from the "non-destructive" version of `AddMultTranspose`